packages/grant-explorer/src/features/api/gap.ts

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from "react";
+import { Hex } from "viem";
+
+const indexerUrl = process.env.REACT_APP_KARMA_GAP_INDEXER_URL;
+
+export interface IGrantStatus {
+  uid: Hex;
+  title: string;
+  text: string;
+  createdAtMs: number;
+}
+
+export interface IGapGrant {
+  uid: Hex;
+  projectUID: Hex;
+  communityUID: Hex;
+  title: string;
+  description: string;
+  createdAtMs: number;
+  milestones: {
+    uid: Hex;
+    title: string;
+    description: string;
+    endsAtMs: number;
+    completed?: IGrantStatus;
+    isGrantUpdate?: boolean;
+  }[];
+  updates: IGrantStatus[];
+}
+
+export function useGap(projectId?: string) {
+  const [grants, setGrants] = useState<IGapGrant[]>([]);
+  const [isGapLoading, setIsGapLoading] = useState(false);
+
+  const getGrantsFor = async (projectRegistryId: string) => {
+    if (!indexerUrl) throw new Error("GAP Indexer url not set.");
+    try {
+      setIsGapLoading(true);
+      const items: IGapGrant[] = await fetch(
+        `${indexerUrl}/grants/external-id/${projectRegistryId}`
+      ).then((res) => res.json());
```

Comment:
Can we use SWR for async data hooks? It will handle caching so revisiting the same projectId will not trigger another network request.
- https://swr.vercel.app

Reply:
I don't think it changed anything around triggering another request but I added swr there.